comp/core/sysprobeconfig/component.go

```diff
@@ -30,5 +33,15 @@ type Component interface {
 	Warnings() *config.Warnings
 
 	// SysProbeObject returns the wrapper sysconfig
-	SysProbeObject() *sysconfig.Config
+	SysProbeObject() *sysconfigtypes.Config
+}
+
+// NewNoneOptional return a None optional type for sysprobeconfig.Component.
```

Comment:
The implementation of module is usually in `impl` folder in order to make `component.go` depends on as less  as possible packages.

Reply:
Yes, but making it part of the `impl` would force code to link with the implementation when not needing the comp (as explain in the rest of the comment).